### PR TITLE
quieter harvesting step

### PR DESCRIPTION
**Pull request by davidlange6**

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: DQMOffline/EGamma/plugins/PhotonDataCertification.cc [hunk 1]
```diff
@@ -85,17 +85,18 @@ float PhotonDataCertification::invMassZtest(string path, TString name,
   if (TestElem == 0) return 0;
   TH1F *TestHist = TestElem->getTH1F();
   if (TestHist == 0) return 0;
+  RooMsgService::instance().setGlobalKillBelow(RooFit::WARNING) ;
```
Comment (Dr15Jones) [inline DQMOffline/EGamma/plugins/PhotonDataCertification.cc]: How thread-safe is that?

Comment (Dr15Jones) [inline DQMOffline/EGamma/plugins/PhotonDataCertification.cc]: Calling `RooMsgService::instance()` is not thread safe
https://root.cern.ch/doc/master/RooMsgService_8cxx_source.html#l00361

Comment (Dr15Jones) [inline DQMOffline/EGamma/plugins/PhotonDataCertification.cc]: It looks like Roo is very thread-hostile. There isn't even the standard ROOT locks anywhere.

**Comment by davidlange6** [comment]
> On May 30, 2017, at 8:25 PM, Chris Jones <notifications@github.com> wrote:
> 
> @Dr15Jones commented on this pull request.
> 
> In DQMOffline/EGamma/plugins/PhotonDataCertification.cc:
> 
> > @@ -85,17 +85,18 @@ float PhotonDataCertification::invMassZtest(string path, TString name,
>    if (TestElem == 0) return 0;
>    TH1F *TestHist = TestElem->getTH1F();
>    if (TestHist == 0) return 0;
> +  RooMsgService::instance().setGlobalKillBelow(RooFit::WARNING) ;
> 
> How thread-safe is that?

no idea. seems unsafe indeed. will see if there is a better way

**Comment by Martin-Grunewald** [comment]
please test

**Comment by dmitrijus** [comment]
+1

File: DQMOffline/EGamma/plugins/PhotonDataCertification.cc (hunk 1 above)
Comment (davidlange6) [inline DQMOffline/EGamma/plugins/PhotonDataCertification.cc]: indeed - looking further, my call is not adding significantly to the # of calls to RooMsgService::instance() - there are already ~10 of them (one per printout line that I'm trying to quiet)

**Pull request merged**